SSA/Projects/InstCombine/AliveAutoGenerated.lean

```diff
@@ -1,11 +1,58 @@
 import SSA.Projects.InstCombine.LLVM.EDSL
-import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.InstCombine.AliveStatements
+import Mathlib.Tactic
 open MLIR AST
 
+open Std (BitVec)
+
+namespace AliveAutoGenerated
+
 abbrev ICom.Refinement (src tgt : Com (φ:=0) Γ t) (h : Goedel.toType t = Option α := by rfl) : Prop :=
-  ∀ Γv, Bitvec.Refinement (h ▸ src.denote Γv) (h ▸ tgt.denote Γv)
+  ∀ Γv, (h ▸ src.denote Γv) ⊑ (h ▸ tgt.denote Γv)
 
-infixr:90 " ⊑ "  => ICom.Refinement
+infixr:90 (priority:=high) " ⊑ "  => ICom.Refinement
+
+/--
+- We first simplify `ICom.refinement` to see the context `Γv`.
+- We `simp_peephole Γv` to simplify context accesses by variables.
+- We simplify the translation overhead.
+- Then we introduce variables, `cases` on the variables to eliminate the `none` cases.
+- We cannot leave it at this state, since then the variables will be inaccessible.
+- So, we revert the variables for the user to re-introduce them as they see fit.
+-/
+macro "simp_alive_peephole" : tactic =>
+  `(tactic|
+      (
+      dsimp only[ICom.Refinement]
+      intros Γv
+      simp_peephole at Γv
+      /- note that we need the `HVector.toPair`, `HVector.toSingle` lemmas since it's used in `InstCombine.Op.denote`
+        We need `HVector.toTuple` since it's used in `MLIR.AST.mkOpExpr`. -/
+      try simp only[OpDenote.denote, InstCombine.Op.denote, HVector.toPair, pairMapM, BitVec.Refinement,
```

Comment:
Add whitespace between only and `[`

Reply:
We also should disable 'decide' in simp using `(config := {decide := false})` as otherwise some theorems will simplify to nothing.